src/sentry/api/endpoints/project_group_index.py

```diff
@@ -546,17 +594,43 @@ def put(self, request, project):
                 res_type = GroupResolution.Type.in_release
                 res_type_str = 'in_release'
                 res_status = GroupResolution.Status.resolved
+            elif statusDetails.get('inCommit'):
+                in_commit = statusDetails['inCommit']
+                commit = in_commit['commit']
+                activity_type = Activity.SET_RESOLVED_IN_COMMIT
+                activity_data = {
+                    'commit': in_commit['commit'].id,
+                }
+                status_details = {
+                    'inCommit': {
+                        'commit': in_commit['commit'].key,
+                        'repository': in_commit['repository'].name,
+                    },
+                    'actor': serialize(extract_lazy_object(request.user), request.user),
+                }
+                res_type_str = 'in_commit'
             else:
-                release = None
                 res_type_str = 'now'
                 activity_type = Activity.SET_RESOLVED
                 activity_data = {}
                 status_details = {}
 
             now = timezone.now()
 
+            if commit and not release:
```

Comment:
My concerns with doing that logic in the endpoint would be the duplication it incurs and the risk of behavior drift with release commit processing. Would another approach be to use post_save hooks on GroupLink to identify if the commit is in an existing release?

Reply:
There is no release commit processing though if its already present. We could do it on post_save hooks but IMO thats no different other than it moves the logic to somewhere else (which would be no different than moving this to another function and calling that function directly). It's also different than the way release commit processing currently works, since you can have multiple releases with a given commit, and this just picks the most recent.

Reply:
Also also, that would make us duplicate the logic of GroupResolution updates here

Reply:
>  It's also different than the way release commit processing currently works, since you can have multiple releases with a given commit, and this just picks the most recent.

I had missed that nuance before, sorry about that.